### PR TITLE
Sign in form

### LINKED ISSUE #28: Sign up and Sign in page
Build sign up and sign in pages, create snapshot and unit test. Connect it to Firebase Auth. Add ability to sign in and sign up using Facebook and Google. Add ability to reset password.

### PR DESCRIPTION
**Pull request by Jamil-Bailony**
- [x] add sign in form: email and password inputs
- [x] style the form
- [x] add validation using Joi resolver
- [x] build unite test for each validation case

## Screenshots
image

image

image

Closes #28

### CODE REVIEW

File: src/components/SignInForm/index.jsx [hunk 1]
```diff
@@ -1,10 +1,80 @@
...
+import InputErrorMessage from "../InputErrorMessage";
+import "./index.scss";
+
+const schema = Joi.object({
+  email: Joi.string()
+    .required()
+    .email({ tlds: { allow: ["com", "net", "edu"] } })
+    .messages({
+      "string.empty": `Email field cannot be empty`,
+      "string.email": `You should type a valid email`,
```
Comment (jennetturkkan) [inline src/components/SignInForm/index.jsx]: "Please provide a valid email" or something in this tone sounds more professional

**Review by jennetturkkan — APPROVED** [review]
Looks good to me

File: src/components/SignInForm/index.jsx [hunk 2]
```diff
@@ -1,10 +1,80 @@
...
+import { joiResolver } from "@hookform/resolvers/joi";
+import InputErrorMessage from "../InputErrorMessage";
+import "./index.scss";
+
+const schema = Joi.object({
+  email: Joi.string()
+    .required()
+    .email({ tlds: { allow: ["com", "net", "edu"] } })
+    .messages({
+      "string.empty": `Email field cannot be empty`,
```
Comment (louisrli) [inline src/components/SignInForm/index.jsx]: Please provide your email

File: src/components/SignInForm/index.jsx [hunk 3]
```diff
@@ -1,10 +1,80 @@
...
+      "string.email": `You should type a valid email`,
+    }),
+  password: Joi.string().required().min(8).max(64).strict().messages({
+    "string.empty": `Password field cannot be empty`,
```
Comment (louisrli) [inline src/components/SignInForm/index.jsx]: Please provide a password

File: src/components/SignInForm/index.test.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  it("should not display error when value is valid", async () => {
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test@mail.com",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/password/i), {
+      target: {
+        value: "str0ngp@$$w0rd",
```
Comment (louisrli) [inline src/components/SignInForm/index.test.js]: this should just be "foo" or something, when you put this here the test readers will think this test is somehow related to a password being strong. by making the string useless you are indicating to readers that it's not relevant to the test

**Review by louisrli — APPROVED** [review]

File: src/components/SignInForm/index.test.js (hunk 4 above)
Comment (Jamil-Bailony) [reply src/components/SignInForm/index.test.js]: yeah, you are right, thank you

**Comment by Jamil-Bailony** [comment]
ping @Ammar-64 : )

**Comment by Ammar-64** [comment]
> ping @Ammar-64 : )

@Jamil-Bailony Can you update the branch please

**Comment by Jamil-Bailony** [comment]
@Ammar-64 done!

**Pull request merged**